lib/src/model/coursetable/course.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter_app/src/model/coursetable/course_period.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+import '../json_init.dart';
+
+part 'course.g.dart';
+
+@JsonSerializable()
+class Course {
```

Comment:
#### For the suggestion of the "prevent putting unneeded instance member into class declaration, or make them as private member.":
 The `Course` class is used to transfer the node from web crewler into data transfer object, it expected to transfer the dirty String into good value and assign into member.

 Since I expected to input the string and set the transfer value into class. I'm not sure how to deal with it, so I add `late` when it will transform into differnt type (`string -> int` or `string -> List<String>` w/ new line split).

 It should be mentioned that `JsonSerializable` seems that not support transfer the constructor with `List<String>` parameter 🤔, therefore the `teacherString` and `teachers` exists.

#### For the suggestion of the "prevent using late":
  I assume it is late binding the data into member. So I add it.

It may need some good implement to optimize the `Course` class but I just can do it as far.

Reply:
Optimized on dfcf849